swayfmt/src/items/item_trait/mod.rs

```diff
@@ -83,13 +83,16 @@ impl Format for ItemTrait {
                     sway_ast::ItemTraitItem::Const(const_decl, _) => {
                         write!(formatted_code, "{}", formatter.indent_str()?,)?;
                         const_decl.format(formatted_code, formatter)?;
-                        writeln!(formatted_code, ";")?;
                     }
                     ItemTraitItem::Error(_, _) => {}
                 }
             }
         }
-        formatted_code.pop(); // pop last ending newline
+
+        if formatted_code.ends_with('\n') {
```

Comment:
This is a good call out. In many parts of the code, it checks only for `\n`. I assumed this was because we convert the newlines into unix newlines during formatting, then convert back at the end. I'll take a closer look to see if that's actually happening.

Reply:
Yes I think that's what's going on. We build the formatted code using `writeln` for new lines, which always inserts unix-style newlines. Then we apply the newline style of the original code at the end.
<img width="782" alt="image" src="https://github.com/FuelLabs/sway/assets/47993817/fb957ebf-ddd9-4213-b8ff-d7961a1ab0af">